src/common/UserProvider/UserProvider.tsx

```diff
@@ -0,0 +1,54 @@
+import React from "react";
+import {
+  User,
+  UserCurrentStatus,
+  UserProviderState,
+  GatekeeperRequestError,
+} from "./interfaces";
+import PropTypes from "prop-types";
+import useSWR from "swr";
+import { fetcher } from "../../libs";
+
+export const UserProviderContext = React.createContext({} as UserProviderState);
+
+/**
+ * Provider that provides user state to any child react element
+ */
+export const UserProvider: React.FC<React.HTMLProps<HTMLDivElement>> = ({
+  children,
+}) => {
+  const response = useSWR<User | GatekeeperRequestError>("/auth/user", fetcher);
+
+  /**
+   * Computes the "status" (UserCurrentStatus) of the user depending on the response from gatekeeper
+   */
+  const status = React.useMemo((): UserCurrentStatus => {
+    if (response.error) {
+      return UserCurrentStatus.Errored;
+    } else if (!response.data) {
+      return UserCurrentStatus.Loading;
+    } else if ((response.data as GatekeeperRequestError).statusCode === 401) {
+      return UserCurrentStatus.LoggedOut;
+    } else {
+      return UserCurrentStatus.LoggedIn;
+    }
+  }, [response.error, response.data]);
+
+  return (
+    <UserProviderContext.Provider
+      value={{
+        user:
+          response.data && (response.data as User).authId
+            ? (response.data as User)
+            : undefined,
```

Comment:
Instead of checking the response, would it be a better idea to use the value of `status` here? Something like:
``` javascript
(status === UserCurrentStatus.LoggedIn) ? (response.data as User) : undefined
```